ldk/javascript/src/promisify.ts

```diff
@@ -0,0 +1,91 @@
+import { Cancellable } from './cancellable';
+
+export function promisify<T>(arg: OliveHelps.Readable<T>): Promise<T> {
+  return new Promise((resolve, reject) => {
+    try {
+      arg((cb: T) => resolve(cb));
+    } catch (e) {
+      console.error(e);
+      reject(e);
+    }
+  });
+}
+
+export function promisifyWithParam<TParam, TOut>(
+  param: TParam,
+  arg: OliveHelps.ReadableWithParam<TParam, TOut>,
+): Promise<TOut> {
+  return new Promise((resolve, reject) => {
+    try {
+      arg(param, (cb: TOut) => resolve(cb));
+    } catch (e) {
+      console.error(e);
+      reject(e);
+    }
+  });
+}
+
+export function promisifyWithTwoParams<TParam1, TParam2, TOut>(
```

Comment:
Some of these are called `<TParamX>` and some are `<TPX>`. I'd be consistent at least in this file.